commands/history/inspect.go

```diff
@@ -0,0 +1,305 @@
+	"strconv"
+	"strings"
+	"text/tabwriter"
+	"time"
+
+	"github.com/containerd/platforms"
+	"github.com/docker/buildx/builder"
+	"github.com/docker/buildx/localstate"
+	"github.com/docker/buildx/util/cobrautil/completion"
+	"github.com/docker/buildx/util/confutil"
+	"github.com/docker/cli/cli/command"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/tonistiigi/go-csvvalue"
+	"google.golang.org/grpc/codes"
+)
+
+type inspectOptions struct {
+	builder string
+	ref     string
+}
+
+func runInspect(ctx context.Context, dockerCli command.Cli, opts inspectOptions) error {
+	b, err := builder.New(dockerCli, builder.WithName(opts.builder))
+	if err != nil {
+		return err
+	}
+
+	nodes, err := b.LoadNodes(ctx)
+	if err != nil {
+		return err
+	}
+	for _, node := range nodes {
+		if node.Err != nil {
+			return node.Err
+		}
+	}
+
+	recs, err := queryRecords(ctx, opts.ref, nodes)
+	if err != nil {
+		return err
+	}
+
+	if len(recs) == 0 {
+		if opts.ref == "" {
+			return errors.New("no records found")
+		}
+		return errors.Errorf("no record found for ref %q", opts.ref)
+	}
+
+	if opts.ref == "" {
+		slices.SortFunc(recs, func(a, b historyRecord) int {
+			return b.CreatedAt.AsTime().Compare(a.CreatedAt.AsTime())
+		})
+	}
+
+	rec := &recs[0]
+
+	ls, err := localstate.New(confutil.NewConfig(dockerCli))
+	if err != nil {
+		return err
+	}
+	st, _ := ls.ReadRef(rec.node.Builder, rec.node.Name, rec.Ref)
+
+	log.Printf("rec %+v", rec)
+	log.Printf("st %+v", st)
+
+	tw := tabwriter.NewWriter(dockerCli.Out(), 1, 8, 1, '\t', 0)
+
+	attrs := rec.FrontendAttrs
+	delete(attrs, "frontend.caps")
+
+	writeAttr := func(k, name string, f func(v string) (string, bool)) {
+		if v, ok := attrs[k]; ok {
+			if f != nil {
+				v, ok = f(v)
+			}
+			if ok {
+				fmt.Fprintf(tw, "%s:\t%s\n", name, v)
+			}
+		}
+		delete(attrs, k)
+	}
+
+	var context string
+	var dockerfile string
+	if st != nil {
+		context = st.LocalPath
+		dockerfile = st.DockerfilePath
+		wd, _ := os.Getwd()
+
+		if dockerfile != "" && dockerfile != "-" {
+			if rel, err := filepath.Rel(context, dockerfile); err == nil {
+				dockerfile = rel
+			}
+		}
+		if context != "" {
+			if rel, err := filepath.Rel(wd, context); err == nil {
+				context = rel
+			}
+		}
+	}
+
+	if v, ok := attrs["context"]; ok && context == "" {
+		delete(attrs, "context")
+		context = v
+	}
+	if dockerfile == "" {
+		if v, ok := attrs["filename"]; ok {
+			dockerfile = v
+			if dfdir, ok := attrs["vcs:localdir:dockerfile"]; ok {
+				dockerfile = filepath.Join(dfdir, dockerfile)
+			}
+		}
+	}
+	delete(attrs, "filename")
+
+	if context != "" {
+		fmt.Fprintf(tw, "Context:\t%s\n", context)
+	}
+	if dockerfile != "" {
+		fmt.Fprintf(tw, "Dockerfile:\t%s\n", dockerfile)
+	}
+	if _, ok := attrs["context"]; !ok {
+		if src, ok := attrs["vcs:source"]; ok {
+			fmt.Fprintf(tw, "VCS Repository:\t%s\n", src)
+		}
+		if rev, ok := attrs["vcs:revision"]; ok {
+			fmt.Fprintf(tw, "VCS Revision:\t%s\n", rev)
+		}
+	}
```

Comment:
I think we could keep the well-known git context format like:

```
VCS: https://github.com/tonistiigi/buildx#dfdd89d50e60af52129a88b47792c9784470f708
```

instead of:

```
VCS Repository: https://github.com/tonistiigi/buildx
VCS Revision:   dfdd89d50e60af52129a88b47792c9784470f708
```